# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1437541263 14400
#      Wed Jul 22 01:01:03 2015 -0400
# Node ID 3c70050faa1eca8bd622e1164b928f2a2963cb02
# Parent  c5343519df789cbd8a9ba5f05325874e17111b93
Fix install paths of fixed and generated test files

* libinterp/module.mk (libinterptestsdir): Delete redundant libinterp.
* liboctave/module.mk (liboctavetestsdir): Delete redundant liboctave.
* test/module.mk (TEST_INST_FILES): New variable renamed from
$(nobase_fixedtests_DATA).  (install-data-local, uninstall-local,
install-test-files, install-test-files): New custom targets to install
fixed test files to correct path.

diff --git a/libinterp/module.mk b/libinterp/module.mk
--- a/libinterp/module.mk
+++ b/libinterp/module.mk
@@ -231,17 +231,17 @@ DLL_CXXDEFS = @OCTINTERP_DLL_DEFS@
 	mv $@-t $@
 
 ## Rules to build test files
 
 LIBINTERP_TST_FILES_SRC := $(shell $(top_srcdir)/build-aux/find-files-with-tests.sh "$(srcdir)" $(ULT_DIST_SRC) $(DLDFCN_SRC))
 
 LIBINTERP_TST_FILES := $(addsuffix -tst,$(LIBINTERP_TST_FILES_SRC))
 
-libinterptestsdir := $(octtestsdir)/libinterp
+libinterptestsdir := $(octtestsdir)
 
 nobase_libinterptests_DATA = $(LIBINTERP_TST_FILES)
 
 ## Override Automake's rule that forces a .hh extension on us even
 ## though we don't want it.  It would be super awesome if automake
 ## would allow users to choose the header file extension.
 .yy.cc:
 	$(AM_V_YACC)$(am__skipyacc) $(SHELL) $(YLWRAP) $< y.tab.c $@ y.tab.h $*.h y.output $*.output -- $(YACCCOMPILE)
diff --git a/liboctave/module.mk b/liboctave/module.mk
--- a/liboctave/module.mk
+++ b/liboctave/module.mk
@@ -94,17 +94,17 @@ LIBOCTAVE_TST_SRC = \
   $(liboctave_system_libsystem_la_SOURCES) \
   $(liboctave_util_libutil_la_SOURCES) \
   $(LIBOCTAVE_TEMPLATE_SRC)
 
 LIBOCTAVE_TST_FILES_SRC := $(shell $(top_srcdir)/build-aux/find-files-with-tests.sh "$(srcdir)" $(LIBOCTAVE_TST_SRC))
 
 LIBOCTAVE_TST_FILES := $(addsuffix -tst,$(LIBOCTAVE_TST_FILES_SRC))
 
-liboctavetestsdir := $(octtestsdir)/liboctave
+liboctavetestsdir := $(octtestsdir)
 
 nobase_liboctavetests_DATA = $(LIBOCTAVE_TST_FILES)
 
 EXTRA_DIST += $(liboctave_EXTRA_DIST)
 
 liboctave_DISTCLEANFILES += \
   $(BUILT_INCS) \
   $(LIBOCTAVE_TST_FILES)
diff --git a/test/module.mk b/test/module.mk
--- a/test/module.mk
+++ b/test/module.mk
@@ -84,21 +84,41 @@ test/bc-overload-tests.stamp: test/build
 	touch $@
 
 GENERATED_TEST_FILES = \
   test/sparse.tst \
   test/bc-overload-tests.stamp
 
 fixedtestsdir := $(octtestsdir)/fixed
 
-nobase_fixedtests_DATA = \
+TEST_INST_FILES = \
   test/sparse.tst \
   $(GENERATED_BC_OVERLOADS_FILES) \
   $(filter-out test/fntests.m, $(TEST_FILES))
 
+install-data-local: install-test-files
+
+uninstall-local: uninstall-test-files
+
+install-test-files:
+	for f in $(TEST_INST_FILES); do \
+	  if test -f "$$f"; then d=; else d="$(srcdir)/"; fi; \
+	  base=`echo $$f | $(SED) 's,^test/,,'`; \
+	  $(MKDIR_P) $(DESTDIR)$(fixedtestsdir)/`echo $$base | $(SED) 's,/*[^/]*$$,,'`; \
+	  $(INSTALL_DATA) $$d$$f $(DESTDIR)$(fixedtestsdir)/$$base; \
+	done
+.PHONY: install-test-files
+
+uninstall-test-files:
+	for f in $(TEST_INST_FILES); do \
+	  base=`echo $$f | $(SED) 's,^test/,,'`; \
+	  rm -f $(DESTDIR)$(fixedtestsdir)/$$base; \
+	done
+.PHONY: uninstall-test-files
+
 BUILT_SOURCES += $(GENERATED_TEST_FILES)
 
 test_EXTRA_DIST += \
   test/build-sparse-tests.sh \
   test/build-bc-overload-tests.sh \
   test/bc-overloads-expected \
   test/build_bc_overloads_expected.m \
   $(TEST_FILES)
